### PR TITLE
Add missing types for imap-simple 4.2

### PR DESCRIPTION
**Pull request by iaarnio**
Add type definitions for functions:
- addBox
- delBox
- getBoxes
- append

Fix parameter name for delFlags.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by iaarnio** [comment]
Type definitions for https://github.com/chadxz/imap-simple

File: types/imap-simple/index.d.ts [hunk 1]
```diff
@@ -42,6 +43,18 @@ export class ImapSimple extends NodeJS.EventEmitter {
...
+    addBox(boxName: string, callback: (err: Error, boxName: string) => void): void;
+    addBox(boxName: string): Promise<string>;
+
+    /** Delete a mailbox, calling the provided callback with signature (err, boxName), or resolves the returned promise with boxName. */
+    delBox(boxName: string, callback: (err: Error, boxName: string) => void): void;
+    delBox(boxName: string): Promise<string>;
+
+    /** Returns the full list of mailboxes (folders). Upon success, either the provided callback will be called with signature (err, boxes), or the returned promise will be resolved with boxes. Boxes is the exact object returned from the node-imap getBoxes() result. */
+    getBoxes(callback: (err: Error, boxes: any) => void): void;
+    getBoxes(): Promise<any>;
```
Comment (pushplay) [inline types/imap-simple/index.d.ts]: Reading the docs it looks like the returned object can be typed `Imap.MailBoxes`.

File: types/imap-simple/index.d.ts [hunk 2]
```diff
@@ -57,6 +70,10 @@ export class ImapSimple extends NodeJS.EventEmitter {
     addMessageLabel(source: string | string[], label: string | string[], callback: (err: Error) => void): void;
     addMessageLabel(source: string | string[], label: string | string[]): Promise<void>;
 
+    /** Appends the argument message to the currently open mailbox or another mailbox. Message is a RFC-822 compatible MIME message. Valid options are mailbox, flags and date. When completed, either calls the provided callback with signature (err), or resolves the returned promise. */
+    append(message: any, options: any, callback: (err: Error) => void): void;
+    append(message: any, options: any): Promise<void>;
```
Comment (pushplay) [inline types/imap-simple/index.d.ts]: Looks like `options` is type `Imap.AppendOptions`.

**Review by pushplay — CHANGES_REQUESTED** [review]
Thanks for updating this.  There are a couple opportunities to improve it further and it's good to go.

**Comment by iaarnio** [comment]
Good observations! I added those more detailed type definitions.

**Review by pushplay — APPROVED** [review]
Cheers.

**Pull request merged**